shared/teams/emojis/add-emoji.tsx

```diff
@@ -0,0 +1,597 @@
+import * as React from 'react'
+import * as TeamsTypes from '../../constants/types/teams'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+import * as RouteTreeGen from '../../actions/route-tree-gen'
+import * as RPCChatGen from '../../constants/types/rpc-chat-gen'
+import * as Container from '../../util/container'
+import * as FsTypes from '../../constants/types/fs'
+import * as ChatTypes from '../../constants/types/chat2'
+import * as ChatConstants from '../../constants/chat2'
+import {pluralize} from '../../util/string'
+import useRPC from '../../util/use-rpc'
+import pickFiles from '../../util/pick-files'
+import kebabCase from 'lodash/kebabCase'
+
+const pickEmojisPromise = () => pickFiles('Select emoji images to upload')
+
+type Props = {
+  conversationIDKey: ChatTypes.ConversationIDKey
+  teamID: TeamsTypes.TeamID // not supported yet
+}
+type RoutableProps = Container.RouteProps<Props>
+
+const filePathToDefaultAlias = (path: string) => {
+  const name = FsTypes.getLocalPathName(path)
+  const lastDot = name.lastIndexOf('.')
+  return kebabCase(lastDot > 0 ? name.slice(0, lastDot) : name)
```

Comment:
what about if it starts with .?

Reply:
That’s fine — this is just to strip the extension. The dot will be converted by kebabCase anyways